src/atomate2/abinit/sets/base.py

```diff
@@ -0,0 +1,564 @@
+                cls._tmpcls_params_set.add(kwarg)
+        return super().__new__(cls)
+
+    def __post_init__(self):
+        """Post init setting of the parameters explicitly set.
+
+        The temporary class variable is also reset (see __new__).
+        """
+        self._params_set = set(self._tmpcls_params_set)
+        self.__class__._tmpcls_params_set = set()
+
+    def __setattr__(self, key, value):
+        """Set a given attribute and register that it is explicitly set."""
+        if key in self.params:
+            self._params_set.add(key)
+        super().__setattr__(key, value)
+
+    def _get_parameters(self, kwargs, prev_generator):
+        params = {}
+        for param in self.params:
+            # If the parameter is in the kwargs, take the value from there.
+            if param in kwargs:
+                val = kwargs[param]
+            else:
+                # If there is no previous generator, take the value from this
+                # generator directly.
+                if prev_generator is None:
+                    val = self.__getattribute__(param)
+                else:
+                    # If the parameter is explicitly set in this generator,
+                    # use that value. Otherwise, take the value from the previous
+                    # generator.
+                    if param in self._params_set:
+                        val = self.__getattribute__(param)
+                    else:
+                        val = prev_generator.__getattribute__(param)
+            params[param] = val
+        # Take extra_abivars from the previous generator if available.
+        extra_abivars = (
+            prev_generator.extra_abivars if prev_generator is not None else {}
+        )
+        # Update extra_abivars from this generator.
+        extra_abivars.update(self.extra_abivars)
+        # Update extra_abivars from kwargs if applicable.
+        extra_abivars.update(kwargs.get("extra_abivars", {}))
+        return params, extra_abivars
+
+    def _get_generator(self, gen_params, extra_abivars):
+        generator = copy.copy(self)
+        generator._params_set = set()
+        for param in self.params:
+            if param in gen_params:
+                generator.__setattr__(param, gen_params[param])
+        generator.extra_abivars = extra_abivars
+        return generator
+
+    def param_is_explicitly_set(self, param):
+        """Check if a given parameter has been explicitly set.
+
+        Parameters
+        ----------
+        param : str
+            Name of parameter.
+
+        Raises
+        ------
+        LookupError
+            If the parameter is not registered.
+        """
+        if param not in self.params:
+            raise LookupError(f'Parameter "{param}" is not registered.')
+        return param in self._params_set
+
+    def get_input_set(  # type: ignore
+        self,
+        structure=None,
+        restart_from=None,
+        prev_outputs=None,
+        **kwargs,
```

Comment:
It is to allow tuning per calculation

Reply:
Yes that's right. The parameters were not passed down. Now they are. I think it can be useful to allow further tuning on a per calculation basis. What do you think ?